src/dos/cdrom_image.cpp

```diff
@@ -1174,6 +1174,235 @@ static std::string dirname(char* file)
 }
 #endif
 
+// Simplified version of an MDS parser.
+// Does not handle copy protection.
+// Credit to reverse engineering efforts by cdemu/libmirage.
+// Code is not taken directly from the project but I read their code:
+// https://github.com/cdemu/cdemu/blob/master/libmirage/images/image-mds/parser.c
+bool CDROM_Interface_Image::LoadMdsFile(const char *mds_filename)
+{
+	const std_fs::path mds_path(to_native_path(mds_filename));
+	std::ifstream file(mds_path, std::ios::in | std::ios::binary);
+
+	// Trailing 1 is a version field.
+	// We're only handling version 1 which is produced by Alcohol 120%
+	// It's the most common and the only one I've come across.
+	constexpr char MdsSignature[] = {'M', 'E', 'D', 'I', 'A', ' ', 'D', 'E', 'S', 'C', 'R', 'I', 'P', 'T', 'O', 'R', 1};
+	char sig[sizeof(MdsSignature)] = {};
+	file.read(sig, sizeof(MdsSignature));
+	if (file.fail()) {
+		return false;
+	}
+	if (memcmp(sig, MdsSignature, sizeof(MdsSignature))) {
+		// Not as MDS file. Fall through to CUE/ISO handlers.
+		return false;
+	}
```

Comment:
How about using `std::array<uint8_t, 17>` here for `MdsSignature` and `sig`? Would be no need to use `sizeof` or `memcmp` the.

Reply:
`std::array` doesn't really help because there's no == overload between that and a C-style array. I'm reading the signature as part of the header struct now and we're using C-style arrays in most of our structs by convention.

`std::string` doesn't work as it's not null terminated.

Reply:
Yeah this is fine.